exp/controllers/rosaroleconfig_controller.go

```diff
@@ -0,0 +1,660 @@
+	}
+
+	runtime.Creator, err = runtime.AWSClient.GetCreator()
+	if err != nil {
+		return err
+	}
+
+	config := roleConfig.Spec.OperatorRoleConfig
+	version := roleConfig.Spec.AccountRoleConfig.Version
+	hostedCp := true
+	forcePolicyCreation := true
+
+	operatorRoles, err := runtime.AWSClient.ListOperatorRoles(version, "", config.Prefix)
+
+	if err != nil {
+		return err
+	}
+
+	for _, roles := range operatorRoles {
+		for _, role := range roles {
+			roleSuffix := strings.TrimPrefix(role.RoleName, config.Prefix)
+			if roleSuffix == role.RoleName {
+				continue
+			}
+			switch roleSuffix {
+			case expinfrav1.IngressOperatorARNSuffix:
+				scope.RosaRoleConfig.Status.OperatorRolesRef.IngressARN = role.RoleARN
+			case expinfrav1.ImageRegistryARNSuffix:
+				scope.RosaRoleConfig.Status.OperatorRolesRef.ImageRegistryARN = role.RoleARN
+			case expinfrav1.StorageARNSuffix:
+				scope.RosaRoleConfig.Status.OperatorRolesRef.StorageARN = role.RoleARN
+			case expinfrav1.NetworkARNSuffix:
+				scope.RosaRoleConfig.Status.OperatorRolesRef.NetworkARN = role.RoleARN
+			case expinfrav1.KubeCloudControllerARNSuffix:
+				scope.RosaRoleConfig.Status.OperatorRolesRef.KubeCloudControllerARN = role.RoleARN
+			case expinfrav1.NodePoolManagementARNSuffix:
+				scope.RosaRoleConfig.Status.OperatorRolesRef.NodePoolManagementARN = role.RoleARN
+			case expinfrav1.ControlPlaneOperatorARNSuffix:
+				scope.RosaRoleConfig.Status.OperatorRolesRef.ControlPlaneOperatorARN = role.RoleARN
+			case expinfrav1.KMSProviderARNSuffix:
+				scope.RosaRoleConfig.Status.OperatorRolesRef.KMSProviderARN = role.RoleARN
+			default:
+				return fmt.Errorf("unknown role suffix: when listing operator roles %s", roleSuffix)
+			}
+		}
+	}
+
+	if !r.operatorRolesReady(&scope.RosaRoleConfig.Status.OperatorRolesRef) {
+		err = operatorroles.CreateOperatorRoles(runtime, ocm.Production, config.PermissionsBoundaryARN, interactive.ModeAuto, policies, version, config.SharedVPCConfig.IsSharedVPC(), config.Prefix, hostedCp, installerRoleArn, forcePolicyCreation,
+			oidcConfigID, config.SharedVPCConfig.RouteRoleARN, ocm.DefaultChannelGroup, config.SharedVPCConfig.VPCEndpointRoleARN)
+		return err
+	}
+
+	return nil
+}
+
+func (r *ROSARoleConfigReconciler) reconcileOIDCConfig(roleConfig *expinfrav1.ROSARoleConfig, scope *scope.RosaRoleConfigScope, ocmClient *ocm.Client) error {
+	if scope.RosaRoleConfig.Status.OIDCID != "" {
+		return nil
+	}
+	if roleConfig.Spec.OperatorRoleConfig.OIDCID != "" {
+		scope.RosaRoleConfig.Status.OIDCID = roleConfig.Spec.OperatorRoleConfig.OIDCID
+		return nil
+	}
+	// Try to get OIDC UUID from some operator role policy document.
+	roleName := fmt.Sprintf("%s-openshift-ingress-operator-cloud-credentials", roleConfig.Spec.OperatorRoleConfig.Prefix)
+	roleDetails, err := scope.IAMClient().GetRole(context.TODO(), &iamv2.GetRoleInput{
+		RoleName: &roleName,
+	})
+	if err != nil {
+		return r.createOIDCConfig(scope, ocmClient)
+	}
```

Comment:
@PanSpagetka the logic here not make sense , if there is an error we don't re-create the oidc config we return the error.